Unblock react native support when using corev2 packages

chradek opened the pull request:
~This is a WIP. Currently these changes, along with a shims file, allows our packages that depend on core-client to run in react native.~

~ToDo: I've created expo and react-native cli projects that illustrate what shims are needed to get this working. Currently working on getting those samples into this PR.~

This PR has been updated to just include making the changes needed for the corev2 packages to run in react native, assuming that the requisite browser APIs are pollyfilled.

ramya-rao-a commented on sdk/core/core-rest-pipeline/src/util/userAgentPlatform.native.ts:
```diff
@@ -0,0 +1,23 @@
+// Licensed under the MIT license.
+
+/*
+ * NOTE: When moving this file, please update "browser" section in package.json.
```
browser -> react-native?

chradek commented on sdk/core/core-http/package.json:
```diff
@@ -56,6 +56,9 @@
     "./dist-esm/src/util/inspect.js": "./dist-esm/src/util/inspect.browser.js",
     "./dist-esm/src/util/url.js": "./dist-esm/src/util/url.browser.js"
   },
+  "react-native": {
```
Remove!

ramya-rao-a replied:
Since react native support will be only part of core-v2, should we add the mapping here?

ramya-rao-a commented:
Build is failing due to formatting changes

HarshaNalluru commented on sdk/core/core-rest-pipeline/package.json:
```diff
@@ -14,6 +14,10 @@
     "./dist-esm/src/util/url.js": "./dist-esm/src/util/url.browser.js",
     "./dist-esm/src/util/userAgentPlatform.js": "./dist-esm/src/util/userAgentPlatform.browser.js"
   },
+  "react-native": {
```
changelog?

deyaaeldeen commented on sdk/core/core-client-paging-rest/package.json:
```diff
@@ -9,6 +9,9 @@
   "browser": {
     "./dist-esm/src/url.js": "./dist-esm/src/url.browser.js"
   },
+  "react-native": {
+    "./dist/index.js": "./dist-esm/src/index.js"
+  },
```
@joheredi is this package still a dep in the rest clients? if not, we could probably skip updating it. Same applies for the lro one.

deyaaeldeen commented on sdk/core/core-client/CHANGELOG.md:
```diff
@@ -10,6 +10,8 @@
 
 ### Other Changes
 
+- Updates package to work with the react native bundler. Browser APIs may still need to be pollyfilled for this package to run in react native. [PR #17783](https://github.com/Azure/azure-sdk-for-js/pull/17783)
+
```
I am curious, is there a way to document those APIs with mappings to the appropriate polyfills?

chradek replied:
The pollyfill landscape in react native is kind of still like the wild west. I plan to have an example shims file in the samples that are included in a follow-up PR, but I don't want to recommend any specific 3rd party pollyfill at this time since I haven't properly vetted them beyond ensuring some seem to work.

deyaaeldeen replied:
That is fair. In order to make it easier for react native customers to use the SDK, should we consider providing documentation of non-standard definitions that need to be polyfilled for every package?

chradek replied:
My plan was to document that in react-native samples (that live alongside our current react/electron samples.) Happy to include some documentation at that point as well similar to how we have documentation on bundling.

deyaaeldeen replied:
Sounds great! Thanks!

deyaaeldeen commented on sdk/core/core-paging/CHANGELOG.md:
```diff
@@ -1,5 +1,11 @@
+
+### Other Changes
+
+- Updates package to work with the react native bundler. Browser APIs such as `URL` will still need to be pollyfilled for this package to run in react native. [PR #17783](https://github.com/Azure/azure-sdk-for-js/pull/17783)
+
```
I believe this package does not reference `URL` or any browser APIs.

chradek replied:
You're right, I was thinking of `core-paging-rest`

chradek commented:
I believe I've addressed all the feedback, let me know if there are any concerns! The potential negative impact here is extremely small since only the react native bundler will read these new mappings!

deyaaeldeen reviewed (approved):
🚢
